### PR TITLE
[android] Fix some tests and a doc flag

### PR DESCRIPTION
**Pull request by finagolfin**
Use `off_t` for a recently failing stdlib test, hack the Python script for some tests so that it works with newer Python, and correct build flag in the docs.

@CodaFi, normally I'd ask @drodriguez but I haven't been able to get a hold of him lately, so would you review?

### CODE REVIEW

File: docs/Android.md [hunk 1]
```diff
@@ -161,5 +161,5 @@ $ utils/build-script \
   --android \                        # Build for Android.
   --android-ndk ~/android-ndk-r23b \  # Path to an Android NDK.
   --android-arch armv7 \             # Optionally specify Android architecture, alternately aarch64
-  --android-ndk-version 21
```
Comment (buttaface) [inline docs/Android.md:164]: There is no such flag, which [was corrected in the build script but missed in this doc](https://github.com/apple/swift/pull/20805/files#diff-9a1726f5e9615fbcad51fcaead5b79cb33f26e6685f4871910e1582ae9049147R257).

File: test/stdlib/POSIX.swift [hunk 2]
```diff
@@ -246,13 +246,7 @@ POSIXTests.test("fcntl(CInt, CInt, UnsafeMutableRawPointer): locking and unlocki
   // Lock for reading...
   var flck = flock()
   flck.l_type = Int16(F_RDLCK)
-  #if os(Android)
-  // In Android l_len is __kernel_off_t which is not the same size as off_t in
-  // 64 bits.
-  flck.l_len = __kernel_off_t(data.utf8.count)
-  #else
   flck.l_len = off_t(data.utf8.count)
```
Comment (buttaface) [inline test/stdlib/POSIX.swift:249]: I think the above comment was written before AArch64 support was added, `off_t` actually works fine. This recently broke on Android AArch64 after [an import was added to this file](https://github.com/apple/swift/pull/39591/files#diff-d653aa83a4c23b7858c2e8a84ff427e6e30dca6138928eb2f06808bbd310a12fR7), with the compiler now complaining that `__kernel_off_t` is an `Int` and the count is an `Int64`, but this change fixes it.

File: utils/swift_build_sdk_interfaces.py [hunk 3]
```diff
@@ -419,6 +419,14 @@ def main():
...
         import threading
         shared_output_lock = threading.Lock()
+        # This is a hack, that was added for newer Python versions to again
+        # work around the lack of sem_open.
+        def thread_lock(self):
+            import threading
+            return threading.Lock()
+        ctx = multiprocessing.get_context()
+        ctxType = type(ctx)
+        ctxType.Lock = thread_lock
```
Comment (buttaface) [inline utils/swift_build_sdk_interfaces.py:429]: [I added the surrounding code to have Android use a ThreadPool instead a couple years ago](https://github.com/apple/swift/pull/27167/files#diff-8da7eca5f7cd5b1d30b731e9c79221c2cb47ac645f1be3dbe5e131c01faf74b9R410) but that also broke with a recent Python 3 version, so I rummaged around in ThreadPool internals till I came up with this hack to get it to run.

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci python lint

**Comment by buttaface** [comment]
Rebased and fixed the Python formatting issue the linter is complaining about, Distributed test failure is unrelated.

**Comment by artemcm** [comment]
@swift-ci please test

File: utils/swift_build_sdk_interfaces.py [hunk 4]
```diff
@@ -410,18 +410,9 @@ def main():
...
-        # Android doesn't support Python's multiprocessing as it doesn't have
-        # sem_open, so switch to a ThreadPool instead.
-        import threading
-        shared_output_lock = threading.Lock()
-        from multiprocessing.pool import ThreadPool
-        pool = ThreadPool(args.jobs, set_up_child,
-                          (args, shared_output_lock))
+    shared_output_lock = multiprocessing.Lock()
+    pool = multiprocessing.Pool(args.jobs, set_up_child,
+                                (args, shared_output_lock))
```
Comment (buttaface) [inline utils/swift_build_sdk_interfaces.py:415]: Someone recently ported `sem_open` to Termux, termux/termux-packages#8993, so go the other way and remove the Android-specific code instead, now that this works.

**Comment by buttaface** [comment]
@MaxDesiatov, would you run the CI on this?

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@buttaface I highly recommend requesting CI access, sending that request shouldn't take much of your time.

**Comment by buttaface** [comment]
@MaxDesiatov, I know, you recommended that before but I'm winding down submitting my Android patches (this is the last patch I apply locally when building trunk on my phone for testing that is worth submitting), and I think it's better not to have CI access once I go into a low-commit mode from here on out (I'm down to [only four trunk pulls pending](https://github.com/search?q=org%3Aapple+author%3Abuttaface&state=open&type=Issues), with plans to submit only one of [my remaining patches](https://github.com/buttaface/termux-packages/tree/swift-trunk/packages/swift)), for security and other reasons.

**Comment by MaxDesiatov** [comment]
I don't see much problem in requesting CI access, and then requesting it to be revoked after you're done with your contributions.

**Comment by buttaface** [comment]
I'm almost out the door, probably not worth the hassle for all involved.

**Pull request merged**